lib/internal/test_runner/test.js

```diff
@@ -714,68 +735,78 @@ class Test extends AsyncResource {
 
     let stopPromise;
 
-    try {
-      if (this.parent?.hooks.before.length > 0) {
-        // This hook usually runs immediately, we need to wait for it to finish
-        await this.parent.runHook('before', this.parent.getRunArgs());
-      }
-      if (this.parent?.hooks.beforeEach.length > 0 && !this.skipped) {
-        await this.parent.runHook('beforeEach', hookArgs);
-      }
-      stopPromise = stopTest(this.timeout, this.signal);
-      const runArgs = ArrayPrototypeSlice(args);
-      ArrayPrototypeUnshift(runArgs, this.fn, ctx);
-
-      if (this.fn.length === runArgs.length - 1) {
-        // This test is using legacy Node.js error first callbacks.
-        const { promise, cb } = createDeferredCallback();
-
-        ArrayPrototypePush(runArgs, cb);
-        const ret = ReflectApply(this.runInAsyncScope, this, runArgs);
-
-        if (isPromise(ret)) {
-          this.fail(new ERR_TEST_FAILURE(
-            'passed a callback but also returned a Promise',
-            kCallbackAndPromisePresent,
-          ));
-          await SafePromiseRace([ret, stopPromise]);
+    while (true) {
```

Comment:
Using an infinite loop here makes me nervous. Would a `for` loop work:

```js
for (this.currentAttempt = 1; this.currentAttempt <= this.retries + 1; this.currentAttempt++)
```

You could still break out of the loop if the test passes.